src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/JintLuceneDocumentConverter.cs

```diff
@@ -172,27 +227,36 @@ protected override int GetFields<T>(T instance, LazyStringValue key, LazyStringV
 
                             BoostDocument(instance, numberOfCreatedFields, documentBoost);
 
-                            continue;
+                            shouldProcessAsBlittable = false;
+                            return;
                         }
                     }
                 }
 
-                value = TypeConverter.ToBlittableSupportedType(actualValue, flattenArrays: false, forIndexing: true, engine: documentToProcess.Engine, context: indexContext);
-                numberOfCreatedFields = GetRegularFields(instance, field, CreateValueForIndexing(value, propertyBoost), indexContext, out _);
-
-                newFields += numberOfCreatedFields;
-
-                BoostDocument(instance, numberOfCreatedFields, documentBoost);
+                shouldProcessAsBlittable = true;
+            }
 
-                if (value is IDisposable toDispose)
-                {
-                    // the value was converted to a lucene field and isn't needed anymore
-                    toDispose.Dispose();
-                }
+            int ProcessAsJson(JsValue actualValue, IndexField field, float? propertyBoost)
+            {
+                var value = TypeConverter.ToBlittableSupportedType(actualValue, flattenArrays: false, forIndexing: true, engine: documentToProcess.Engine,
+                    context: indexContext);
+                return GetRegularFields(instance, field, CreateValueForIndexing(value, propertyBoost), indexContext, out _);
             }
+            
+            bool IsDynamicFieldMap(JsValue propertyDescriptorValue, string propertyAsString, IndexField field, out IEnumerator<JsValue> iterator)
```

Comment:
IsDynamicFieldMap -> IsDynamicFieldEnumerable or IsDynamicFieldArray

Reply:
Well, now this function is also checking if a field is spatial so the name should be more generic.